test/Common/DurableTaskLifeCycleNotificationTest.cs

```diff
@@ -1243,15 +1243,16 @@ public void EventGridApiConfigureCheck()
 
             var wrappedOptions = new OptionsWrapper<DurableTaskOptions>(options);
             var connectionStringResolver = new TestConnectionStringResolver();
+            var serviceFactory = new AzureStorageDurabilityProviderFactory(
+                    wrappedOptions,
+                    connectionStringResolver,
+                    mockNameResolver.Object,
+                    NullLoggerFactory.Instance);
             var extension = new DurableTaskExtension(
                 wrappedOptions,
                 new LoggerFactory(),
                 mockNameResolver.Object,
-                new AzureStorageDurabilityProviderFactory(
-                    wrappedOptions,
-                    connectionStringResolver,
-                    mockNameResolver.Object,
-                    NullLoggerFactory.Instance),
+                new[] { serviceFactory }.ToList(),

```

Comment:
No need for `.ToList()` here.
```suggestion
                new[] { serviceFactory },
```